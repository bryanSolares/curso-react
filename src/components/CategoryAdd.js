import React, { useState } from "react";
import PropTypes from "prop-types";

export const CategoryAdd = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState("");

  const handleInputChange = (event) => {
    setInputValue(event.target.value);
  };

  const handleSubmit = (event) => {
    event.preventDefault();

    if (inputValue.trim().length > 2) {
      setCategories((categories) => [inputValue, ...categories]);
      setInputValue("");
    }
  };

  return (
    <>
      <h2>Category Add</h2>
      <form onSubmit={handleSubmit}>
        <input type="text" value={inputValue} onChange={handleInputChange} />
      </form>
    </>
  );
};

CategoryAdd.propTypes = {
  setCategories: PropTypes.func.isRequired,
};
