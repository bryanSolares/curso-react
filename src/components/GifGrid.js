import React from "react";
import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifGridItem } from "./GifGridItem";

export const GifGrid = ({ category }) => {
  //   const [images, setImages] = useState([]);
  //   useEffect(() => {
  //     getGifs(category).then(setImages);
  //   }, [category]); // esto se implemento porque cada vez que el componente cambiaba se necesitaba recargar, se traslada a custome hook

  const { data: images, loading } = useFetchGifs(category);

  return (
    <>
      <h3 className="animate__animated animate__fadeIn"> {category} </h3>

      {loading && <p className="animate__animated animate__flash">Cargando...</p>}

      <div className="card-grid">
        {images.map((img) => (
          <GifGridItem key={img.id} {...img} />
        ))}
      </div>
    </>
  );
};
