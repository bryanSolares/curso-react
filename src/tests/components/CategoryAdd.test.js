import { shallow, ShallowWrapper } from "enzyme";
import { CategoryAdd } from "../../components/CategoryAdd";

describe("Pruebas en el componente <CategoryAdd/>", () => {
  const setCategories = jest.fn();
  let wrapper;

  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallow(<CategoryAdd setCategories={setCategories} />);
  });

  test("Debe mostrar el componente correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe cambiar la caja de texto", () => {
    const input = wrapper.find("input");
    const value = "Hola Mundo";
    input.simulate("change", { target: { value } });
    const parrafo = wrapper.find("p").text().trim();
    expect(parrafo).toBe(value);
  });

  test("No debe postear la informacion OnSubmit", () => {
    wrapper.find("form").simulate("submit", { preventDefault() {} });
    expect(setCategories).not.toHaveBeenCalled();
  });

  test("Debe de llamar al setCategories y limpiar la caja de texto", () => {
    const input = wrapper.find("input");
    input.simulate("change", { target: { value: "Hola Mundo" } });
    wrapper.find("form").simulate("submit", { preventDefault() {} });
    expect(setCategories).toHaveBeenCalled();
    expect(setCategories).toHaveBeenCalledTimes(1);
    expect(setCategories).toHaveBeenCalledWith(expect.any(Function));
    expect(input.prop("value")).toBe("");
  });
});
