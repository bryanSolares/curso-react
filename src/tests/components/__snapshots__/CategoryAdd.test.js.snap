// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <CategoryAdd/> Debe mostrar el componente correctamente 1`] = `
<Fragment>
  <p>
     
     
  </p>
  <h2>
    Category Add
  </h2>
  <form
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      type="text"
      value=""
    />
  </form>
</Fragment>
`;
